### ISSUE TITLE
TypeError: 'HistoryIterator' object is not an iterator

### ISSUE THREAD

**Issue by Vthechamp22** [python-discord/sir-lancebot#493]
## Description
When running the `seasonalbot`, I occasionally get an error:
```py
Ignoring exception in on_reaction_add
Traceback (most recent call last):
  File "~\.virtualenvs\seasonalbot-9qy_Ovb8\lib\site-packages\discord\client.py", line 312, in _run_event
    await coro(*args, **kwargs)
  File "seasonalbot\bot\utils\decorators.py", line 79, in guarded_listener
    return await listener(*args, **kwargs)
  File "seasonalbot\bot\exts\halloween\candy_collection.py", line 76, in on_reaction_add
    if message.id in await self.ten_recent_msg():
  File "seasonalbot\bot\exts\halloween\candy_collection.py", line 138, in ten_recent_msg
    msg = await next(channel.history(limit=1, before=o))
TypeError: 'HistoryIterator' object is not an iterator
```
This repeats several [times](https://paste.pythondiscord.com/ehokihoqad.sql)

## Steps to Reproduce
Run the `seasonalbot` using `pipenv run start`, and wait for some time

## Expected Behaviour

No error to occur and `candy_collections.py` to run normally

## Actual Behaviour

Errrror

## Known Impacted Platforms

- [ ] Web
- [ ] Desktop
- [ ] Android App
- [ ] iOS App
*It didn't impact anything, but an error is an error right?*

## Would you like to implement a fix?

***Note: For high-priority or critical bugs, fixes may be implemented by staff.***

- [ ] I'd like to implement the bug fix
- [X] Anyone can implement the bug fix

**Comment by Vthechamp22** [comment]
EDIT: I am not sure if this is a bug for everyone, but it is happening for me so I thought I should mention it

**Comment by SebastiaanZ** [comment]
It should be `await channel.history(limit=1, before=o).next()`.

**Comment by quanta-kt** [comment]
I have removed these lines in #496 
I assume I can make a note about this in my PR saying `"Closes #493"` ?

**Comment by Vthechamp22** [comment]
I think this issue has been closed by PR #496
